_examples/rest/main.go

```diff
@@ -140,33 +144,40 @@ func ArticleCtx(next http.Handler) http.Handler {
 // It's just a stub, but you get the idea.
 func SearchArticles(w http.ResponseWriter, r *http.Request) {
 	// Filter by query param, and search...
-	render.JSON(w, r, articles)
+	render.Respond(w, r, articles)
 }
 
 // ListArticles returns an array of Articles.
 func ListArticles(w http.ResponseWriter, r *http.Request) {
-	render.JSON(w, r, articles)
+	render.Respond(w, r, articles)
 }
 
 // CreateArticle persists the posted Article and returns it
 // back to the client as an acknowledgement.
 func CreateArticle(w http.ResponseWriter, r *http.Request) {
-	var data struct {
-		*Article
-		OmitID interface{} `json:"id,omitempty"` // prevents 'id' from being set
-	}
-	// ^ the above is a nifty trick for how to omit fields during json unmarshalling
-	// through struct composition
-
-	if err := render.Bind(r.Body, &data); err != nil {
-		render.JSON(w, r, err.Error())
+	data := &ArticleRequest{}
+	if err := render.Decode(r, &data); err != nil {
+		render.Respond(w, r, err.Error())
 		return
 	}
 
 	article := data.Article
 	dbNewArticle(article)
 
-	render.JSON(w, r, article)
+	render.Status(r, http.StatusCreated)
+	render.Respond(w, r, article)
+}
+
+// Experiment using render.Bind() that automatically decodes the request
+// body and makes a new object.
+func CreateArticle2(w http.ResponseWriter, r *http.Request) {
+	data := render.RequestBody(r.Context()).(*ArticleRequest)
```

Comment:
yea, this is why I don't like it, and going to delete it from the PR -- but I wanted to show you guys to prototype before I did that